retrofit-converters/moshi/src/test/java/retrofit/MoshiConverterTest.java

```diff
@@ -51,45 +49,75 @@
     }
   }
 
-  static class ExampleAdapter {
-    @ToJson public void to(JsonWriter writer, Example example) throws IOException {
-      writer.beginObject();
-      writer.name("name").value(example.getName());
-      writer.endObject();
+  static class AnInterfaceAdapter {
+    @ToJson public void write(JsonWriter jsonWriter, AnInterface anInterface) throws IOException {
+      System.out.println("TO JSON: " + anInterface);
+      jsonWriter.beginObject();
+      jsonWriter.name("name").value(anInterface.getName());
+      jsonWriter.endObject();
     }
 
-    @FromJson public Example from(JsonReader reader) throws IOException {
-      throw new UnsupportedOperationException(); // Moshi requires this method to exist.
+    @FromJson public AnInterface read(JsonReader jsonReader) throws IOException {
+      jsonReader.beginObject();
+
+      String name = null;
+      while (jsonReader.peek() != JsonReader.Token.END_OBJECT) {
```

Comment:
`while (jsonReader.hasNext())`

... Unless you're going for symmetry with the other libraries!